docs/guide.rst

```diff
@@ -300,4 +300,117 @@ You can read more about how cairo treats shortstrings in `the documentation <htt
 Conversion functions and references:
 
 - :obj:`encode_shortstring <starknet_py.cairo.felt.encode_shortstring>`
-- :obj:`decode_shortstring <starknet_py.cairo.felt.decode_shortstring>`
\ No newline at end of file
+- :obj:`decode_shortstring <starknet_py.cairo.felt.decode_shortstring>`
+
+
+
+L1 <> L2 messaging
+------------------
+
+To retrieve the L1 -> L2 or L2 -> L1 message count, you need to provide some data that you used to create that message.
+Then after creating the message's representation, you can query it's current count.
+
+You can find out more about L1 <> L2 messaging here: https://starknet.io/documentation/l1-l2-messaging/
+
+Full API description :ref:`here<Messaging>`.
+
+
+
+L1 -> L2 messages
+#################
+
+The message's count is an `int`, representing the number of unconsumed messages on L2 with that exact content.
+Since the `nonce`'s value will always be unique for each message, this value is either 0 or 1
+(0 meaning the message is consumed or not received yet, and 1 for unconsumed, queued message).
+
+.. code-block:: python
+
+    from starknet_py.net.l1.messages import (
+        L2Message,
+        L2MessageContent,
+    )
+    from starknet_py.net.models import StarknetChainId
+
+        ## All of the construction methods shown below are correct:
+
+        # 1. From message content
+        l1_to_l2_msg = L1ToL2Message.from_content(
+            L1ToL2MessageContent(
+                l1_sender=123, # Integer representation of L1 hex address
+                l2_recipient="0x123123123", # Either a hex L2 address, or it's integer representation
+                nonce=1, # Can be retrieved from L1 transaction's receipt (the one containing the sent message)
+                function_name="dummy_name", # L2 recipient function name
+                payload=[32, 32, 32, 32], # L2 Function calldata, list of ints
+            )
+        )
+
+        # 2. From message hash
+        l1_to_l2_msg = L1ToL2Message.from_hash(
+            (123).to_bytes(32, "big") # Provide 32 bytes as an input here, instead of message's content
+        )
+
+        # 3. From transaction receipt
+        w3 = web3.Web3(web3.providers.HTTPProvider("https://my-rpc-endpoint.com/"))
+        tx_receipt = w3.eth.wait_for_transaction_receipt("0x123123123")
+        l2_to_l1_msg = L2ToL1Message.from_tx_receipt(tx_receipt)
```

Comment:
Please change l2_to_l1_msg to l1_to_l2_msg

same for L2ToL1Message

Same for exmaple 4.